internal/cli/cli.go

```diff
@@ -233,9 +234,63 @@ func documentGet(serverURL string, apiKey string, id string, idIsShareID bool) e
 
 	b, err := json.MarshalIndent(doc, "", "  ")
 	if err != nil {
-		return fmt.Errorf("failed marshalling document with id '%s: %w", id, err)
+		return fmt.Errorf("failed marshalling document with id '%s': %w", id, err)
 	}
 	fmt.Println(string(b))
 
 	return nil
 }
+
+func documentUpdate() *cobra.Command {
+	var title, text string
+	var append, publish bool
+
+	cmd := &cobra.Command{
```

Comment:
Wasn't the goal to NOT have cobra in these functions to make them testable?
Before #50 we had exactly the sturcutre you introduce right now 😁 
See https://github.com/ioki-mobility/go-outline/blob/e73af9c32c6af9a90c338a68ab5106c04a054d62/internal/cli/cli.go

Reply:
Yes that is true, at the time we were adding features like crazy :smile:  and I didn't got myself to look into cobra well enough and how it is used. So we even though we are reverting back, IMO I see that new approach is still testable (well no in the perfect way) and at the same time complexity and verbosity of creating commands gets reduced. So you win some loose some here.

Reply:
Could you maybe provide a sample test? 😇

Reply:
sure, here is what I see:
```shell
go test -v ./internal/cli/
=== RUN   Test
    package_test.go:18: Body: {"id":"doc_id","title":"TITLE"}
        
    package_test.go:20: URL: /api/documents.update
    package_test.go:22: Headers: map[Accept:[application/json] Accept-Encoding:[gzip] Authorization:[Bearer KKEEEYY] Content-Length:[32] Content-Type:[application/json] User-Agent:[Go-http-client/1.1]]
null
--- PASS: Test (0.00s)
PASS
ok 
```

After applying following changes:
```diff
diff --git a/client.go b/client.go
index 3abfefd..905ce8f 100644
--- a/client.go
+++ b/client.go
@@ -3,8 +3,8 @@ package outline
 import (
 	"net/http"
 
-	"github.com/rsjethani/rsling"
 	"github.com/ioki-mobility/go-outline/internal/common"
+	"github.com/rsjethani/rsling"
 )
 
 // Client is per server top level client which acts as entry point and stores common configuration (like base url) for
diff --git a/internal/cli/package_test.go b/internal/cli/package_test.go
new file mode 100644
index 0000000..a9214e7
--- /dev/null
+++ b/internal/cli/package_test.go
@@ -0,0 +1,32 @@
+package cli
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Right now we are just logging but we can obviously verify against expected values.
+
+		b, _ := io.ReadAll(r.Body)
+		t.Logf("Body: %v\n", string(b))
+
+		t.Logf("URL: %v\n", r.URL.String())
+
+		t.Logf("Headers: %v\n", r.Header)
+
+	}))
+	defer srv.Close()
+
+	os.Args = append(os.Args, "document", "update", "--key", "KKEEEYY",
+		"--server", srv.URL, "--title", "TITLE", "doc_id")
+
+	err := Run()
+	require.NoError(t, err)
+}
```
Of course some tweaks to cli code would be needed for better organization and reliable tests.